### PR TITLE
Add package ocaml-sfml version 0.08

### PR DESCRIPTION
**Pull request by fccm**

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
```
#=== ERROR while compiling ocaml-sfml.0.08 ====================================#
# context              2.0.5 | linux/x86_64 | ocaml-base-compiler.4.08.1 | pinned(https://github.com/fccm/ocaml-sfml/archive/v0.08.tar.gz)
# path                 ~/.opam/4.08/.opam-switch/build/ocaml-sfml.0.08
# command              /usr/bin/make -C src cxx_all
# exit-code            2
# env-file             ~/.opam/log/ocaml-sfml-15-ece7a3.env
# output-file          ~/.opam/log/ocaml-sfml-15-ece7a3.out
### output ###
# make: Entering directory '/home/opam/.opam/4.08/.opam-switch/build/ocaml-sfml.0.08/src'
# ocamldep SF*.ml SF*.mli oo_*.ml oo_*.mli > ml_depend.mk
# g++ -g -Wall -Werror -Wno-deprecated-declarations -fPIC -c -I/home/opam/.opam/4.08/lib/ocaml -I/usr/include/ ./cxx_stubs/sf_caml_conv.cpp
# g++ -g -Wall -Werror -Wno-deprecated-declarations -fPIC -c -I/home/opam/.opam/4.08/lib/ocaml -I/usr/include/ ./cxx_stubs/sf_conv_vectors.cpp
# ./cxx_stubs/sf_conv_vectors.cpp:26:10: fatal error: SFML/System/Vector2.hpp: No such file or directory
#  #include <SFML/System/Vector2.hpp>
#           ^~~~~~~~~~~~~~~~~~~~~~~~~
# compilation terminated.
# make: *** [Makefile:180: sf_conv_vectors.o] Error 1
# make: Leaving directory '/home/opam/.opam/4.08/.opam-switch/build/ocaml-sfml.0.08/src'

- make: Entering directory '/home/opam/.opam/4.08/.opam-switch/build/ocaml-sfml.0.08/src'
- ocamldep SF*.ml SF*.mli oo_*.ml oo_*.mli > ml_depend.mk
- g++ -g -Wall -Werror -Wno-deprecated-declarations -fPIC -c -I/home/opam/.opam/4.08/lib/ocaml -I/usr/include/ ./cxx_stubs/sf_caml_conv.cpp
- g++ -g -Wall -Werror -Wno-deprecated-declarations -fPIC -c -I/home/opam/.opam/4.08/lib/ocaml -I/usr/include/ ./cxx_stubs/sf_conv_vectors.cpp
- ./cxx_stubs/sf_conv_vectors.cpp:26:10: fatal error: SFML/System/Vector2.hpp: No such file or directory
-  #include <SFML/System/Vector2.hpp>
-           ^~~~~~~~~~~~~~~~~~~~~~~~~
- compilation terminated.
- make: *** [Makefile:180: sf_conv_vectors.o] Error 1
- make: Leaving directory '/home/opam/.opam/4.08/.opam-switch/build/ocaml-sfml.0.08/src'
```
Is this an expected error?

File: packages/ocaml-sfml/ocaml-sfml.0.08/opam [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+maintainer: "Florent Monnier <monnier.florent@gmail.com>"
+authors: ["Florent Monnier"]
+
+homepage: "https://github.com/fccm/ocaml-sfml"
+dev-repo: "git+https://github.com/fccm/ocaml-sfml.git"
+bug-reports: "https://github.com/fccm/ocaml-sfml/issues"
+doc: "http://www.linux-nantes.org/~fmonnier/ocaml/sfml/doc/"
+
+license: "zlib/png"
+synopsis: "Bindings to the SFML multimedia library."
```
Comment (kit-ty-kate) [inline packages/ocaml-sfml/ocaml-sfml.0.08/opam]:
```suggestion
synopsis: "Bindings to the SFML multimedia library"
```

**Comment by fccm** [comment]
> Is this an expected error?

You need to install the SFML lib first.
On my computer with mageia linux I installed libsfml-devel and libsfml2.5 which provide SFML 2.5.1
The -devel package provides the include files that are located in:
/usr/include/SFML/
So the error:
fatal error: SFML/System/Vector2.hpp: No such file
You should have the following file on your system:
/usr/include/SFML/System/Vector2.hpp
Oops, I just see that I put the wrong conf- in the depends field, maybe it will resolve the problem.

**Comment by kit-ty-kate** [comment]
Please don't use `-warn-error` in releases or this will happen at every new release of OCaml:
```
# ocamlopt -g -w A -warn-error A -ccopt "-Wall -Werror -Wno-deprecated-declarations" -c -o SFFont.cmx SFFont.ml
# File "SFFont.ml", line 32, characters 0-51:
# 32 | external getInfo: u -> info = "caml_sfFont_getInfo"
#      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# Error (warning 61): This primitive declaration uses type info, which is unannotated and
# unboxable. The representation of such types may change in future
# versions. You should annotate the declaration of info with [@@boxed]
# or [@@unboxed].
# make: *** [Makefile:170: SFFont.cmx] Error 2
# make: Leaving directory '/home/opam/.opam/4.08/.opam-switch/build/ocaml-sfml.0.08/src'

<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build ocaml-sfml 0.08
```

**Comment by fccm** [comment]
Should I also remove -Werror from the -ccopt of ocamlopt?

**Comment by kit-ty-kate** [comment]
> Should I also remove -Werror from the -ccopt of ocamlopt?

Yes, this is advisable.

**Comment by kit-ty-kate** [comment]
```
# + /usr/bin/c++ -O3 '-fvisibility=hidden' -fPIC -I/home/opam/.opam/4.08/.opam-switch/build/ocsfml.2.0/camlpp '-std=c++0x' -fpermissive -c -I/usr/include -I/usr/include -I/home/opam/.opam/4.08/lib/ocaml -I../OcsfmlSystem/ocsfml_system_stub -I../OcsfmlWindow/ocsfml_window_stub -I../OcsfmlGraphics/ocsfml_graphics_stub OcsfmlGraphics/ocsfml_graphics_stub/Font.cpp -oOcsfmlGraphics/ocsfml_graphics_stub/Font.o
# In file included from /home/opam/.opam/4.08/.opam-switch/build/ocsfml.2.0/camlpp/camlpp/custom_class.hpp:5,
#                  from ../OcsfmlGraphics/ocsfml_graphics_stub/Font.hpp:4,
#                  from OcsfmlGraphics/ocsfml_graphics_stub/Font.cpp:1:
# /usr/include/c++/8/functional: In instantiation of 'struct std::_Bind_check_arity<const sf::Glyph& (sf::Font::*)(unsigned int, unsigned int, bool, float) const, const std::_Placeholder<1>&, const std::_Placeholder<2>&, const std::_Placeholder<3>&, const std::_Placeholder<4>&>':
# /usr/include/c++/8/functional:816:12:   required from 'struct std::_Bindres_helper<const sf::Glyph&, const sf::Glyph& (sf::Font::*)(unsigned int, unsigned int, bool, float) const, const std::_Placeholder<1>&, const std::_Placeholder<2>&, const std::_Placeholder<3>&, const std::_Placeholder<4>&>'
# /usr/include/c++/8/functional:832:5:   required by substitution of 'template<class _Result, class _Func, class ... _BoundArgs> typename std::_Bindres_helper<_Result, _Func, _BoundArgs>::type std::bind(_Func&&, _BoundArgs&& ...) [with _Result = const sf::Glyph&; _Func = const sf::Glyph& (sf::Font::*)(unsigned int, unsigned int, bool, float) const; _BoundArgs = {const std::_Placeholder<1>&, const std::_Placeholder<2>&, const std::_Placeholder<3>&, const std::_Placeholder<4>&}]'
# OcsfmlGraphics/ocsfml_graphics_stub/Font.cpp:30:3:   required from here
# /usr/include/c++/8/functional:775:7: error: static assertion failed: Wrong number of arguments for pointer-to-member
#        static_assert(_Varargs::value
#                      ~~~~~~~~~~~~~~~
#        ? sizeof...(_BoundArgs) >= _Arity::value + 1
#        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#        : sizeof...(_BoundArgs) == _Arity::value + 1,
#        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# /usr/include/c++/8/functional: In instantiation of 'std::_Bind_result<_Result, _Functor(_Bound_args ...)>::__disable_if_void<_Res> std::_Bind_result<_Result, _Functor(_Bound_args ...)>::__call(std::tuple<_Args2 ...>&&, std::_Index_tuple<_Indexes ...>) [with _Res = const sf::Glyph&; _Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}; long unsigned int ..._Indexes = {0, 1, 2, 3}; _Result = const sf::Glyph&; _Functor = const sf::Glyph& (sf::Font::*)(unsigned int, unsigned int, bool, float) const; _Bound_args = {std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>}; std::_Bind_result<_Result, _Functor(_Bound_args ...)>::__disable_if_void<_Res> = const sf::Glyph&]':
# /usr/include/c++/8/functional:651:24:   required from 'std::_Bind_result<_Result, _Functor(_Bound_args ...)>::result_type std::_Bind_result<_Result, _Functor(_Bound_args ...)>::operator()(_Args&& ...) [with _Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}; _Result = const sf::Glyph&; _Functor = const sf::Glyph& (sf::Font::*)(unsigned int, unsigned int, bool, float) const; _Bound_args = {std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>}; std::_Bind_result<_Result, _Functor(_Bound_args ...)>::result_type = const sf::Glyph&]'
# /home/opam/.opam/4.08/.opam-switch/build/ocsfml.2.0/camlpp/camlpp/res_management.hpp:62:15:   required from 'T camlpp::res_management<T, flags>::call_helper(Func&&, Args&& ...) [with Func = std::_Bind_result<const sf::Glyph&, const sf::Glyph& (sf::Font::*(std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>))(unsigned int, unsigned int, bool, float) const>; Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}; T = const sf::Glyph&; int flags = 0]'
# /home/opam/.opam/4.08/.opam-switch/build/ocsfml.2.0/camlpp/camlpp/res_management.hpp:82:40:   required from 'void camlpp::res_management<T, flags>::call_exception_flag(std::false_type, value&, Func&&, Args&& ...) [with Func = std::_Bind_result<const sf::Glyph&, const sf::Glyph& (sf::Font::*(std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>))(unsigned int, unsigned int, bool, float) const>; Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}; T = const sf::Glyph&; int flags = 0; std::false_type = std::integral_constant<bool, false>; value = long int]'
# /home/opam/.opam/4.08/.opam-switch/build/ocsfml.2.0/camlpp/camlpp/res_management.hpp:89:7:   required from 'void camlpp::res_management<T, flags>::call(value&, Func&&, Args&& ...) [with Func = std::_Bind_result<const sf::Glyph&, const sf::Glyph& (sf::Font::*(std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>))(unsigned int, unsigned int, bool, float) const>; Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}; T = const sf::Glyph&; int flags = 0; value = long int]'
# OcsfmlGraphics/ocsfml_graphics_stub/Font.cpp:30:3:   required from here
# /usr/include/c++/8/functional:556:24: error: no matching function for call to '__invoke(const sf::Glyph& (sf::Font::*&)(unsigned int, unsigned int, bool, float) const, const sf::Font&, unsigned int&, unsigned int&, bool&)'
#     return std::__invoke(_M_f, _Mu<_Bound_args>()
#            ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
#          (std::get<_Indexes>(_M_bound_args), __args)...);
#          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# In file included from /usr/include/c++/8/tuple:41,
#                  from /usr/include/c++/8/functional:54,
#                  from /home/opam/.opam/4.08/.opam-switch/build/ocsfml.2.0/camlpp/camlpp/custom_class.hpp:5,
#                  from ../OcsfmlGraphics/ocsfml_graphics_stub/Font.hpp:4,
#                  from OcsfmlGraphics/ocsfml_graphics_stub/Font.cpp:1:
# /usr/include/c++/8/bits/invoke.h:89:5: note: candidate: 'template<class _Callable, class ... _Args> constexpr typename std::__invoke_result<_Functor, _ArgTypes>::type std::__invoke(_Callable&&, _Args&& ...)'
#      __invoke(_Callable&& __fn, _Args&&... __args)
#      ^~~~~~~~
# /usr/include/c++/8/bits/invoke.h:89:5: note:   template argument deduction/substitution failed:
# /usr/include/c++/8/bits/invoke.h: In substitution of 'template<class _Callable, class ... _Args> constexpr typename std::__invoke_result<_Functor, _ArgTypes>::type std::__invoke(_Callable&&, _Args&& ...) [with _Callable = const sf::Glyph& (sf::Font::*&)(unsigned int, unsigned int, bool, float) const; _Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}]':
# /usr/include/c++/8/functional:556:24:   required from 'std::_Bind_result<_Result, _Functor(_Bound_args ...)>::__disable_if_void<_Res> std::_Bind_result<_Result, _Functor(_Bound_args ...)>::__call(std::tuple<_Args2 ...>&&, std::_Index_tuple<_Indexes ...>) [with _Res = const sf::Glyph&; _Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}; long unsigned int ..._Indexes = {0, 1, 2, 3}; _Result = const sf::Glyph&; _Functor = const sf::Glyph& (sf::Font::*)(unsigned int, unsigned int, bool, float) const; _Bound_args = {std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>}; std::_Bind_result<_Result, _Functor(_Bound_args ...)>::__disable_if_void<_Res> = const sf::Glyph&]'
# /usr/include/c++/8/functional:651:24:   required from 'std::_Bind_result<_Result, _Functor(_Bound_args ...)>::result_type std::_Bind_result<_Result, _Functor(_Bound_args ...)>::operator()(_Args&& ...) [with _Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}; _Result = const sf::Glyph&; _Functor = const sf::Glyph& (sf::Font::*)(unsigned int, unsigned int, bool, float) const; _Bound_args = {std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>}; std::_Bind_result<_Result, _Functor(_Bound_args ...)>::result_type = const sf::Glyph&]'
# /home/opam/.opam/4.08/.opam-switch/build/ocsfml.2.0/camlpp/camlpp/res_management.hpp:62:15:   required from 'T camlpp::res_management<T, flags>::call_helper(Func&&, Args&& ...) [with Func = std::_Bind_result<const sf::Glyph&, const sf::Glyph& (sf::Font::*(std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>))(unsigned int, unsigned int, bool, float) const>; Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}; T = const sf::Glyph&; int flags = 0]'
# /home/opam/.opam/4.08/.opam-switch/build/ocsfml.2.0/camlpp/camlpp/res_management.hpp:82:40:   required from 'void camlpp::res_management<T, flags>::call_exception_flag(std::false_type, value&, Func&&, Args&& ...) [with Func = std::_Bind_result<const sf::Glyph&, const sf::Glyph& (sf::Font::*(std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>))(unsigned int, unsigned int, bool, float) const>; Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}; T = const sf::Glyph&; int flags = 0; std::false_type = std::integral_constant<bool, false>; value = long int]'
# /home/opam/.opam/4.08/.opam-switch/build/ocsfml.2.0/camlpp/camlpp/res_management.hpp:89:7:   required from 'void camlpp::res_management<T, flags>::call(value&, Func&&, Args&& ...) [with Func = std::_Bind_result<const sf::Glyph&, const sf::Glyph& (sf::Font::*(std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>))(unsigned int, unsigned int, bool, float) const>; Args = {const sf::Font&, unsigned int&, unsigned int&, bool&}; T = const sf::Glyph&; int flags = 0; value = long int]'
# OcsfmlGraphics/ocsfml_graphics_stub/Font.cpp:30:3:   required from here
# /usr/include/c++/8/bits/invoke.h:89:5: error: no type named 'type' in 'struct std::__invoke_result<const sf::Glyph& (sf::Font::*&)(unsigned int, unsigned int, bool, float) const, const sf::Font&, unsigned int&, unsigned int&, bool&>'
# Command exited with code 1.
# make[2]: *** [CMakeFiles/ocsfml.dir/build.make:61: CMakeFiles/ocsfml] Error 10
# make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/ocsfml.dir/all] Error 2
# make: *** [Makefile:84: all] Error 2
```
There are still some problems with the C++ binding. The version used in Debian stable is 2.5.1, so theoretically it should be the same the same as your local computer but for some reason it fails here.

**Comment by kit-ty-kate** [comment]
To debug it you can use docker as follow:
```
$ docker run --rm -it ocaml/opam2
```

**Comment by fccm** [comment]
Sorry but this is an error: I'm not the maintainer of ocsfml, I'm only willing to provide a package for ocaml-sfml. When you told me to remove the dot at the end of the synopsis, I also removed the dot at the end of the synopsis of ocsfml because both are bindings for SFML and when I made the opam file of ocaml-sfml I used the one from ocsfml as a template to start with.

I should probably not have edited this other package, is there a way to fix the situation?
Should we close this PR and make a new one?

**Comment by kit-ty-kate** [comment]
my bad I didn't see it wasn't one of the new package. My commit to undo your change should fix the issue and let the CI check the rest

**Comment by kit-ty-kate** [comment]
There is a bunch of errors witg `ocaml-sfml.0.08` on OCaml < 4.06:
For instance, on 4.02:
```
# g++ -g -Wall -Werror -Wno-deprecated-declarations -fPIC -c -I/home/opam/.opam/4.02/lib/ocaml -I/usr/include/ ./cxx_stubs/SFCursor_stub.cpp
# In file included from ./cxx_stubs/sf_caml_incs.hpp:9,
#                  from ./cxx_stubs/SFCursor_stub.cpp:26:
# ./cxx_stubs/SFCursor_stub.cpp: In function 'value caml_sfCursor_loadFromPixels(value, value, value)':
# /home/opam/.opam/4.02/lib/ocaml/caml/memory.h:88:29: error: unused variable 'caml__frame' [-Werror=unused-variable]
#    struct caml__roots_block *caml__frame = caml_local_roots
#                              ^~~~~~~~~~~
# /home/opam/.opam/4.02/lib/ocaml/caml/memory.h:99:3: note: in expansion of macro 'CAMLparam0'
#    CAMLparam0 (); \
#    ^~~~~~~~~~
# ./cxx_stubs/SFCursor_stub.cpp:40:2: note: in expansion of macro 'CAMLparam3'
#   CAMLparam3(data, size, hotspot);
#   ^~~~~~~~~~
# ./cxx_stubs/SFCursor_stub.cpp: In function 'value caml_sfCursor_loadFromSystem(value)':
# /home/opam/.opam/4.02/lib/ocaml/caml/memory.h:88:29: error: unused variable 'caml__frame' [-Werror=unused-variable]
#    struct caml__roots_block *caml__frame = caml_local_roots
#                              ^~~~~~~~~~~
# /home/opam/.opam/4.02/lib/ocaml/caml/memory.h:91:3: note: in expansion of macro 'CAMLparam0'
#    CAMLparam0 (); \
#    ^~~~~~~~~~
# ./cxx_stubs/SFCursor_stub.cpp:57:2: note: in expansion of macro 'CAMLparam1'
#   CAMLparam1(type);
#   ^~~~~~~~~~
# cc1plus: all warnings being treated as errors
# make: *** [Makefile:180: SFCursor_stub.o] Error 1
# make: Leaving directory '/home/opam/.opam/4.02/.opam-switch/build/ocaml-sfml.0.08/src'
```
And on 4.05:
```
# g++ -g -Wall -Werror -Wno-deprecated-declarations -fPIC -c -I/home/opam/.opam/4.05/lib/ocaml -I/usr/include/ ./cxx_stubs/SFUdpSocket_stub.cpp
# ./cxx_stubs/SFUdpSocket_stub.cpp: In function 'value caml_sfUdpSocket_receive(value, value)':
# ./cxx_stubs/SFUdpSocket_stub.cpp:96:13: error: 'Bytes_val' was not declared in this scope
#              Bytes_val(data), maxSize, sizeReceived, address, port);
#              ^~~~~~~~~
# ./cxx_stubs/SFUdpSocket_stub.cpp:96:13: note: suggested alternative: 'Class_val'
#              Bytes_val(data), maxSize, sizeReceived, address, port);
#              ^~~~~~~~~
#              Class_val
# make: *** [Makefile:180: SFUdpSocket_stub.o] Error 1
# make: Leaving directory '/home/opam/.opam/4.05/.opam-switch/build/ocaml-sfml.0.08/src'
```
To fix this you can either fix the release or restrict this package for OCaml >= 4.06

**Comment by kit-ty-kate** [comment]
Oh I just realized this is a new package. New packages it is best to apply the rule: no `ocaml-` prefix for the packages. Could you rename your package to `sfml` instead?

**Comment by kit-ty-kate** [comment]
It fails on several distribution due to the lack of depexts and fixed version of external libraries support but it seems ok. We can move from there. Thanks a lot!

**Pull request merged**